agent/consul/state/catalog.go

```diff
@@ -2104,6 +2108,52 @@ func (s *Store) serviceDumpKindTxn(tx *memdb.Txn, ws memdb.WatchSet, kind struct
 	return s.parseCheckServiceNodes(tx, nil, idx, "", results, err)
 }
 
+func (s *Store) IngressGatewaysForService(ws memdb.WatchSet, serviceName string, entMeta *structs.EnterpriseMeta) (uint64, structs.CheckServiceNodes, error) {
```

Comment:
Is it intentional that this is only invoked by tests in this PR?

Reply:
This function is being used in `agent/consul/health_endpoint.go` to grab the ingress gateways for a specific service.

https://github.com/hashicorp/consul/pull/7509/files#diff-3993ba95ffed38e5a49949f12d13156dR293

Reply:
Ah I had already collapsed that file and Ctrl-F didn't reopen it.